### PR TITLE
Removes all the unnecessary pytest.mark.parametrize 

### LINKED ISSUE #142: Remove all the unnecessary pytest.mark.parametrize
🐞 **Problem**

Tests are currently parametrized on single set of values. `pytest.mark.parametrize` has a different essence. Remove the redundant test data from everywhere. Add pytest fixtures wherever required. 
https://docs.pytest.org/en/latest/parametrize.html

🎯 **Goal**

💡 **Possible solutions**

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by Manvi07**
Fixes #142 

This pull request removes redundant parameters using fixtures and the unnecessary parametrize decorators by adding the parameters in the test functions itself. 

@shreyasbapat, kindly review it.

### CODE REVIEW

File: .pytest_cache/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# pytest cache directory #
+
+This directory contains data from the pytest's cache plugin,
+which provides the `--lf` and `--ff` options, as well as the `cache` fixture.
+
+**Do not** commit this to version control.
```
Comment (shreyasbapat) [inline .pytest_cache/README.md]: The line says :

```
**Do not** commit this to version control.

```

File: .pytest_cache/v/cache/lastfailed [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "src/einsteinpy/tests/test_utils/manvi.py::test_maxwell_tensor_contravariant_for_natural_units[1e+22-5.5-1.2566370614359172-45.0-0.7]": true,
+  "src/einsteinpy/tests/test_utils/test_kerr_utils.py::test_scaled_spin_factor_raises_error": true,
+  "src/einsteinpy/tests/test_utils/test_kerrnewman_utils.py::test_maxwell_tensor_contravariant_for_natural_units": true,
+  "src/einsteinpy/tests/test_utils/test_kerrnewman_utils.py::test_maxwell_tensor_contravariant_for_natural_units[1e+22-5.5-1.2566370614359172-45.0-0.7]": true
```
Comment (shreyasbapat) [inline .pytest_cache/v/cache/lastfailed]: This also!

File: .pytest_cache/v/cache/nodeids [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  "src/einsteinpy/tests/test_metric/test_schwarzschild.py::test_calculate_trajectory[pos_vec2-vel_vec2-time2-M2-0.0-0.001-OdeMethodKwargs2]",
+  "src/einsteinpy/tests/test_metric/test_schwarzschild.py::test_calculate_trajectory2",
+  "src/einsteinpy/tests/test_metric/test_schwarzschild.py::test_calculate_trajectory3",
+  "src/einsteinpy/tests/test_metric/test_schwarzschild.py::test_calculate_trajectory_iterator[pos_vec0-vel_vec0-time0-M0-0.0-0.0003-OdeMethodKwargs0-True]",
+  "src/einsteinpy/tests/test_metric/test_schwarzschild.py::test_calculate_trajectory_iterator[pos_vec1-vel_vec1-time1-M1-0.0-0.0004-OdeMethodKwargs1-False]",
+  "src/einsteinpy/tests/test_metric/test_schwarzschild.py::test_calculate_trajectory_iterator_RuntimeWarning",
+  "src/einsteinpy/tests/test_metric/test_schwarzschild.py::test_calculate_trajectory_iterator_RuntimeWarning2",
+  "src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py::test_plot_attractor_is_called_only_once",
+  "src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py::test_plot_calls_plot_attractor",
+  "src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py::test_plot_show_shows_plot",
```
Comment (shreyasbapat) [inline .pytest_cache/v/cache/nodeids]: Neither this!

File: src/einsteinpy/tests/test_metric/test_kerrnewman.py [hunk 4]
```diff
@@ -70,22 +71,21 @@ def test_calculate_trajectory1(M, r, end_lambda, stepsize):
...
-            100 * u.C,
-            200.0,
-            1.0,
-        )
-    ],
-)
-def test_compare_calculate_trajectory_iterator_bl(pos_vec, vel_vec, q, M, a, Q, el, ss):
+@pytest.fixture()
+def test_input():
+    q = 1 * u.C / u.g
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_metric/test_kerrnewman.py]: Good use of fixtures :)

File: src/einsteinpy/tests/test_metric/test_kerrnewman.py [hunk 5]
```diff
@@ -70,22 +71,21 @@ def test_calculate_trajectory1(M, r, end_lambda, stepsize):
...
+    a = 1e-6
+    Q = 100 * u.C
+    el = 200.0
+    ss = 1.0
+    return (q, a, Q, el, ss)
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_metric/test_kerrnewman.py]:
```suggestion
    return q, a, Q, el, ss
```
This will also work :)

**Comment by shreyasbapat** [comment]
Also, @Manvi07 you didn't follow the 7th point mentioned here: https://github.com/einsteinpy/einsteinpy/blob/master/CONTRIBUTING.rst#development-environment

Make sure you create branch for everything you do from the next time :)

**Comment by Manvi07** [comment]
> Also, @Manvi07 you didn't follow the 7th point mentioned here: https://github.com/einsteinpy/einsteinpy/blob/master/CONTRIBUTING.rst#development-environment
> 
> Make sure you create branch for everything you do from the next time :)

Sure! I'll take care.

**Comment by Manvi07** [comment]
@shreyasbapat all the suggested changes have been made. :) 
Codecov still fails, I don't know where the coverage is not satisfied. Please review.

**Comment by shreyasbapat** [comment]
One last comment: 
https://codecov.io/gh/einsteinpy/einsteinpy/compare/0931f66acfe94ca6aef3e2d96cd9d1fa8acb578c...f8d19862083aa3bd4707ea824457fdcac308d24b/changes
Can you see this file and write a test for it?

**Comment by shreyasbapat** [comment]
Thanks a lot @Manvi07 . I think this test is impossible to write as of now :/

**Comment by ritzvik** [comment]
Should't we add `*/.pytest_cache` to `.gitignore` ?

**Pull request merged**